src/rest_framework_api_key/permissions.py

```diff
@@ -18,12 +18,13 @@ def get(self, request: HttpRequest) -> typing.Optional[str]:
 
     def get_from_authorization(self, request: HttpRequest) -> typing.Optional[str]:
         authorization = request.META.get("HTTP_AUTHORIZATION")
```

Comment:
Hmm. Looking at https://www.django-rest-framework.org/api-guide/authentication/#tokenauthentication, I see DRF's `TokenAuthentication` docs use the "keyword" wording as well.

So not fussed about either option.

But those docs make me think of a nice alternative too:

> **Note**: If you want to use a different keyword in the header, such as `Bearer`, simply subclass `TokenAuthentication` and set the `keyword` class variable.

So we could allow doing this…

```python
from rest_framework_api_key.permissions import BaseHasAPIKey, KeyParser

class BearerKeyParser(KeyParser):
    keyword = "Bearer"


class HasAPIKey(BaseHasAPIKey):
    key_parser = BearerKeyParser()

```

The nice thing about this would be that it'd allow people to do more complicated things if need be (eg multiple perm classes with different schemes and what not), as opposed to the setting option with only allows overriding the behavior once and for all (requiring you to customize everything if you need more complex behavior).

So WDYTA that `KeyParser.keyword` option?

(Sorry about the fast think-and-update here, I'm just thinking of alternative ways that'd also take into account extra use cases and longer-term maintenance.)

Reply:
I actually really like this option as it does allow for much more customization. I have made a commit implementing this. Take a look at the latest code. This method also removed the need for the additional unit test I wrote (so i removed it).

Reply:
:+1: Fab